app/svelte/src/client/preview/render.js

```diff
@@ -1,13 +1,30 @@
 import { document } from 'global';
 import { stripIndents } from 'common-tags';
 
+let listeners = [];
+
+/**
+ * Since we may have added events using `.on()` we need to ensure that these
+ * events are cleaned up when changing between stories.
+ *
+ * @see mountView()
+ */
+function cancelPreviousStoryListeners() {
+  listeners.forEach(listener => listener.cancel());
+
```

Comment:
I think wrap these in a `try/catch`.
listeners might be referencing some internal variable.. or DOM node, which got removed when the component was switched before this function could run

Reply:
Calling `.cancel()` seems to just be spliced off the this._handlers array inside the component, if it still exists in that array. Would it still be a problem if the listener was referencing something that was non existent, since that code would not actually run?

Reply:
Actually, you are right. Because events supplied via `on` never actually correspond to actual DOM events.

And in this light, I don't think we need to cleanup these handlers at all.
Because the `_handlers` array will get de-referenced when the component is removed, and will be garbage collected.

However, we will need to call `destroy` on the components. So that they can remove their internal event handlers.
A component might have `<svelte:window on:resize='something()'>` components inside it. 
And these events will linger even after component was destroyed

Reply:
Good hustle.